restli-tools/src/test/resources/idls/curr-greeting-simple-fail.restspec.json

```diff
@@ -1,5 +1,6 @@
 {
   "name" : "greeting",
+  "d2ServiceName": "greetingSimpleD2",
```

Comment:
There are a couple concerns I have with this change:
1. Should the d2 service name be surfaced in the resource class? This seems like a tight coupling that assumes Rest.li resources will be used with d2. That may not always be the case.
2. If we decide that it should be surfaced in the resource class, should it be a first-class citizen on the resource annotation? This seems to imply that this annotation will affect native framework behavior. As far as I can tell, there are no corresponding changes in d2. Perhaps then it would be more appropriate as a custom annotation defined for the D2-specific use case, since it's just inert arbitrary data added onto the resource.

Reply:
custom annotation is a good point

Reply:
1. No other easy way to handle routing in a clean way.
2. Custom annotations are weird because we do want to use this to check backward compatibility in the framework

Discussed offline to update the javadocs to indicate the same.